data/dictionaryHelper.js

```diff
@@ -0,0 +1,60 @@
+const fs = require('fs');
+
+
+/**
+ * Extract gqlSetup object from a dictionary
+ * @param {Object} dict from dictionary.json or whatever: https://domain/api/v0/submission/_dictionary/_all
+ * @return {experimentType, fileTypeList} gqlSetup object used by data/gqlSetup.js
+ */
+function dictToGQLSetup( dict ) {
+  const fileTypeList = Object.keys( dict ).filter( key => typeof dict[key] === "object" && dict[ key ].category === "data_file" );
+  // admin types that link to the 'project' level and are not project or program
+  const adminTypeList = Object.keys( dict ).filter( 
+    key => {
+      const entry = dict[key];
+      return key !== "program" && key !== "project" && typeof entry === "object" && entry.category === "administrative" 
+        && Array.isArray( entry.links ) && entry.links.length > 0;
+    } 
+  ).map(
+    key => {
+      const entry = dict[key];
+      return {
+        typeName: key,
+        entry: entry,
+        projectLink: entry.links.find( link => link.target_type === 'project' && link.required )
+      };
+    }
+  );
+  const adminLookup = adminTypeList.reduce( (db,info) => { db[info.typeName] = info; return db; }, {}  );
```

Comment:
what's adminLookup and adminTypeList used for?

Reply:
nothing
I started out thinking I would try to pick some other admin table if 'experiment' or 'study' aren't present, but then decided to not bother.  Zac's schema doesn't have experiment, study, case, or aliquot.  I'll remove that adminLookup line, but I'll leave the adminTypeList there - the code is still in flux, and it may come in handy ...